File: C++/pi.cpp
// NEED:
// A C++-Compiler with C++11-standard e.g. gcc
// USAGE:
// first type "g++ -std=c++11 ./pi.cpp"
// second run with "./a.out" on Linux or "./a.exe" on Windows 
#include <iostream>
#include <string>
#include <stdlib.h>
#include <cmath>
#include <random>

using namespace std;

int main()
{
  //needed variables to approximate pi
  long iterations;
  double total = 0.0;
  double inCircle = 0.0;
  double radius = 1.0;

  //random generator from C++11
  default_random_engine gen;
  uniform_real_distribution<double> dis(0.0,1.0);
  // get user input, otherwise use default value
  cout << "Type limit of iterations\nIterations: ";
  cin >> iterations;

  // for-loop that generates random points and checks if they are in circle
  double distance;
  for(long i = 0; i < iterations; i++){
    double x = dis(gen) * radius;
    double y = dis(gen) * radius;
    total++;
    distance = sqrt(x*x + y*y);
    if(distance < radius)
      inCircle++;
  }

  //pi approximation
  double pi = 4.0 * (inCircle / total);

  // user output
  cout << "PI-Approximation: " << pi << endl;
  return 0;
}

File: JavaScript/pi.js
// NEED:
// A Browser
// USAGE:
// open index.html in browser

// needed variables to approximate pi
var iterations = 10000;
var total = 0.0;
var inCircle = 0.0;
var radius = 1.0;

function approximatePi(){
  // get user input, otherwise use default value
  var input = document.getElementById('iterations')
  if(input.value != "")
    iterations = Number(input.value);

  // for-loop that generates random point and checks if they are in circle or not
  for(var i = 0; i < iterations; i++){
    var point = { 'x': Math.random()*radius, 'y': Math.random()*radius};
    total++;
    var distance = Math.sqrt(Math.pow(point.x,2) + Math.pow(point.y,2));
    if(distance < radius)
        inCircle++;
  }
  // pi approximation
  var pi = 4 * (inCircle / total)

  // write output
  document.getElementById('approximation').innerHTML = "PI-Approximation: " + pi;
}
